app.js

```diff
@@ -1,100 +1,256 @@
+    console.log("url: " + wegwijsUrl);
 
-    const response = await fetch(`https://api.wegwijs.vlaanderen.be/v1/search/organisations?q=kboNumber:${identifiers.kbo}&fields=ovoNumber,kboNumber)`);
+    const response = await fetch(wegwijsUrl);
     const data = await response.json();
 
-    let wegwijsOvo = null;
-    if (data.length) {
-      // We got a match on the KBO, getting the associated OVO back
-      const wegwijsInfo = data[0]; // Wegwijs should only have only one entry per KBO
-      if (wegwijsInfo.ovoNumber)
-        wegwijsOvo = wegwijsInfo.ovoNumber;
+    let kboObject = null;
+    if (!data.length) {
+      return throwServerError(API_STATUS_CODES.STATUS_402, res);
+    }
+    // We got a match on the KBO, getting the associated OVO back
+    const wegwijsInfo = data[0]; // Wegwijs should only have only one entry per KBO
+    kboObject = getKboFields(wegwijsInfo);
+    const kboIdentifiers = await getKboOrgnizationInfo(identifiers.adminUnit);
+
+    if (!kboIdentifiers && kboObject) {
+      await createKbo(kboObject, identifiers.kboId, identifiers.adminUnit);
     }
 
-    if (wegwijsOvo && (wegwijsOvo != identifiers.ovo)) {
+    if (isUpdate(kboObject, kboIdentifiers)) {
+      updateKboOrg(kboObject, kboIdentifiers);
+    }
+
+    let wegwijsOvo = kboObject.ovoNumber ?? null;
+    //Update Ovo Number
+    if (wegwijsOvo && wegwijsOvo != identifiers.ovo) {
+      console.log(identifiers.ovo);
+
       let ovoStructuredIdUri = identifiers.ovoStructuredId;
 
       if (!ovoStructuredIdUri) {
-        ovoStructuredIdUri = await constructOvoStructure(identifiers.kboStructuredId);
+        ovoStructuredIdUri = await constructOvoStructure(
+          identifiers.kboStructuredId
+        );
       }
-
       await updateOvoNumberAndUri(ovoStructuredIdUri, wegwijsOvo);
     }
 
-    return res.status(200).send(); // since we await, it should be 200
+    return throwServerError(API_STATUS_CODES.STATUS_200, res); // since we await, it should be 200
   } catch (e) {
-    console.log("Something went wrong while calling /sync-from-kbo", e);
-    return res.status(500).send();
+    return throwServerError(API_STATUS_CODES.STATUS_500, res, e);
   }
 });
 
-new CronJob(CRON_PATTERN, async function() {
-  const now = new Date().toISOString();
-  console.log(`OVO numbers healing triggered by cron job at ${now}`);
-  try {
-    await healOvoNumbers();
-  } catch (err) {
-    console.log(`An error occurred during OVO numbers healing at ${now}: ${err}`)
-  }
-}, null, true);
+function getKboFields(data) {
+  const {
+    changeTime,
+    name: organisationName,
+    shortName,
+    ovoNumber,
+    kboNumber,
+    labels = [],
+    contacts = [],
+    organisationClassifications = [],
+    locations = [],
+  } = data;
+
+  //no labels = undefined
+  //formal name according to KBO
+  let formalName = extractObjectData(labels, WEGWIJS_DATA_OBJECT_IDS.LABELS)?.value;
```

Comment:
**Praise:** Elegant solution to get rid of the previously duplicated code.

Reply:
removed uuid